### ISSUE TITLE
autogen.pl problems with PMIx 1.2.3

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#4109]
I was noticing when running autogen.pl on a system that use to be able to autogen correctly for the 2.1.x now fails in part of the pmix configury.  This seems to be related to the upgrade to 1.2.3. For master and the other branches, autogen.pl still works for me.

Unfortunately the system where i observed this problem is not currently accessible.

**Comment by rhc54** [comment]
Do you recall the error message it emitted?

**Comment by hppritcha** [comment]
Okay system is available again.  Here's the error I'm seeing:
```
=== Processing subdir: /home/hpp/ompi/opal/mca/pmix/pmix112/pmix
--- Found autogen.sh; running...
Running: ./autogen.sh
==> Running autoreconf
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I ./config
configure.ac:173: error: AC_PROG_CC cannot be called after AM_PROG_CC_C_O
configure.ac:173: the top level
autom4te: /usr/bin/m4 failed with exit status: 1
aclocal: error: echo failed with exit status: 1
autoreconf: aclocal failed with exit status: 1
Command failed: ./autogen.sh
```
running ./autogen.pl  on master and 3.0.x works.  Google seems to indicate this may be an automake issue.

On the system where autogen.pl is failing, I'm using automake version 1.13.4.

**Comment by hppritcha** [comment]
Per discussions with @jsquyres we'd like for the PMIX configury for 1.2.3 to be modified so that 
one can run Open MPI's autogen.pl using the 

```
# Minimum versions
my $ompi_automake_version = "1.12.2";
my $ompi_autoconf_version = "2.69";
my $ompi_libtool_version = "2.4.2";
```
values in that file.

**Comment by rhc54** [comment]
yeah, we are familiar with that error coming from 1.13 - it actually is an automake bug that was fixed later on. I believe we know the workaround, so I'll take a look with @jjhursey

**Comment by rhc54** [comment]
@hppritcha Can you let us know about #4147? If it fixes the problem for you, then we can proceed with a PMIx v1.2.4 release.

**Comment by jsquyres** [comment]
@nmorey Reported a similar problem on the PR that attempted to fix this issue (#4147).

@nmorey Can you tell us the exact versions of Autoconf, Automake, Libtool, and m4 that you are using?

**Comment by nmorey** [comment]
@jsquyres 
```
[(master) nmorey@portia:openmpi ((dc949ba6dd...))]$ autoconf --version 
autoconf (GNU Autoconf) 2.69

[(master) nmorey@portia:openmpi ((dc949ba6dd...))]$ automake --version 
automake (GNU automake) 1.13.4

[(master) nmorey@portia:openmpi ((dc949ba6dd...))]$ libtool --version
libtool (GNU libtool) 2.4.2

[(master) nmorey@portia:openmpi ((dc949ba6dd...))]$ m4 --version
m4 (GNU M4) 1.4.16
```

**Comment by hppritcha** [comment]
This issue was closed via PR #4196
